### DISCUSSION TITLE
One client reads all messages from all queues?

### DISCUSSION THREAD

**Discussion by ZedZipDev** [rabbitmq/rabbitmq-dotnet-client#1147] [General]
I use queues bound to 'direct' exchange. It was my goal:

I have N clients, each of them connects to RabbitMQ, creates its own queue, bind it to 'direct' exchange,
An app-publisher sends 1 message and it broadcasting to all these  queues. i.e. N queues has N messages: queue has 1
Every customer reads its own queue and consumes messages from the queue only.

I run c# apps and check the RabbitMQ:
I run loop and send K messages for each of N queues
Check the R : ok each of N queue has K messages

Run the reader :
```
  var connection = connectionFactory.CreateConnection();
            var channel = connection.CreateModel();
            channel.BasicQos(0, 1, false);
            var msgReceiverd = new MessageReceiverd(channel, RMQList2);
            channel.BasicConsume(RMQQueueName.Value, false, msgReceiverd);

```
Output is all K*N messages (instead of K)
Check the Web page and see all queues are idle and no messages

Why one client reads all messages from all queues bound to the same direct exchange?

**Comment by sungam3r** [comment]
An answer depends of topology used - exact parameters for `Bind` method and so on.

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
A client (connection or a group of connections within an application) can consume from multiple queues, with one consumer per queue. Generally the recommendation is to use separate connections for publishers and consumers.
